README.txt

Recipe Radar

(originally by Jay Miller [jnmiller@cryptofreak.org])

http://www.cryptofreak.org/projects/reciperadar

$Id: README.txt 1031 2008-03-27 05:41:46Z jnmiller $


Installation
~~~~~~~~~~~~
You've probably already installed the addon.  In case you're not sure how
to do it, unzip the archive to your

   ... \World of Warcraft\Interface\AddOns

directory.  A new directory called 'RecipeRadar' should appear, along with
several files and subdirectories.  That should be all you need to do to
use Recipe Radar.

When you get in game, you may want to map a key to toggle Recipe Radar.
You should be able to find the option to do this in your Key Bindings menu,
all the way at the bottom.  Otherwise, you can hit the new icon on your
minimap to open Recipe Radar or use the slash command, '/rrdr'.


Bugs
~~~~
If you spot a bug in this software, *please let me know!*  The best way
to do so is to hit the bug tracker at

   http://wow.curseforge.com/addons/recipe-radar/tickets/

and leave a short report on what was happening at the time.  If possible,
please include the version of Recipe Radar that caused the bug and the
language of your WoW client.

Also in the bugs category are misplaced vendors.  The location information
for all of the vendors in the Recipe Radar database was retrieved in a 
mostly automated way, so some coordinates are incorrect.  If you come upon
a vendor whose mapped location does not match their actual location, please
let me know!  The most helpful thing you can do is this quick process:

   1) Target the mis-mapped vendor.
   2) Move your character exactly on top of the vendor.
   3) Enter the following slash command on a single line:

      /script SetMapToCurrentZone() local x, y = GetPlayerMapPosition("player")  RecipeRadar_Print(format("'%s': [%.3f, %.3f]", UnitName("target"), x, y))

   4) Record the text you see.  It should resemble this:
   
      'Drac Roughcut': [0.356, 0.489]
   
   5) File a bug report and include this text.

Thank you very much for your help!


Acknowledgements
~~~~~~~~~~~~~~~~
- I'd like to thank Nimrah for passing on Technetium's TGAtoBLP2 code.
  It dropped the almost 50 maps from 343k each to 130k each!
- Esamynn is responsible for Astrolabe, a high-quality, generic mapping
  addon that Recipe Radar now relies on.


Thanks for trying Recipe Radar!  Enjoy!


ChangeLog.txt
Recipe Radar ChangeLog

v1.430; November 29, 2011

- Bump TOC
- Update sublibs

v1.402; October 3, 2011

- Incorporated 4.1 and 4.2 updates from Speedywaystar and SecureAgent
- Updated Astrolabe for Firelands

v1.33; August 11, 2009

- Fixed several "no coordinates" asserts, sigh.

v1.32; August 9, 2009

- Updated for WoW 3.2, including the new recipes.
- Alchemy should work again for German users.
- Fixed a bunch of incorrect recipe data.
- Updated the Icecrown map.

v1.31; April 14, 2009

- Updated for WoW 3.1.

v1.30; March 11, 2009

- Updated recipe data; asserts should no longer appear in Dalaran, Stormwind
  and Orgrimmar.  Also updated the Stormwind map.
- Recipe Radar is now able to parse collapsed reputation sections.
- Corrupted alt databases should be repaired on next use.
- Trade skill filters (eg. searches) should no longer adversely affect your
  character's known recipes.
- This is also true for others' linked trade skills (thank you, Thortok2000!).

v1.29; November 15, 2008

- Fixed major problems with the Russian and German translations.
- Added lots more WotLK recipes.
- Item-based costs now show up explicitly.

v1.28; November 7, 2008

- Updated for WoW 3.0.
- Added Yor's Russian translations. (Thank you!)
- Added some new recipes and vendors, including some WotLK content.

v1.27; April 7, 2008

- Fixed the lockup bug on some zone changes.
- Fixed a bug that triggered when two players have the same profession but
  different specialties.
- Users should no longer see the question mark icon for uncached recipes.
- Adjusted some vendor attributes.

v1.26; March 31, 2008

- Fixed the "'DongleStub' (a nil value)" bug.
- Fixed zhTW recipe parsing.

v1.25; March 28, 2008

- Added new v2.4 recipes.
- Added Shattered Sun Offensive faction content.
- Added support for square and rotating minimaps.
- Fixed Ashtongue Deathsworn reputation parsing.

v1.24; October 10, 2007

- Included the new recipes added in v2.2.
- Tailoring specialists should no longer see recipes from other
  specializations.

v1.23; July 24, 2007

- Major changes to the Spanish and Chinese name maps.
- Fixed some Scryr/Aldor recipe mistakes.
- Specialties now show up in the 'Note' field.
- Added a couple of new recipes.
- Fixed Hillsbrad Foothills on German clients (I hope).

v1.22; April 13, 2007

- Fixed recipe availability checking for Unicode locales.

v1.21; April 12, 2007

- Added a Chinese Traditional translation (thank you, Edeson!).
- Spanish recipes should now be recognized as known, I hope.

v1.20; February 27, 2007

- All of the new factions should now actually work (sigh).
- Removed some troublesome recipe inaccuracies.
- Fixed item linking.
- Localized vendor names.

v1.19; February 5, 2007

- Revised Spanish translation (thank you, Jorge Pozo!).
- Revised French translation (thank you, akirra!).
- Updated and added some vendor/recipe information.
- Winter Veil vendors should now only appear during December (thank
  you, Elkano!).

v1.18; January 27, 2007

- Added all of the known BC recipes and vendors.
- Tailoring should now *really* work for Spanish versions.
  Leatherworking was also broken but should now work (thank you, J!).

v1.17; January 18, 2007

- Fixed a bug that caused an error on startup for some users.
- Tailoring should now work for Spanish versions.

v1.16; January 17, 2007

- Added support for Jewelcrafting.
- Added several new recipes and vendors.
- The options frame is now draggable.
- Removed incorrect (old) specializations.
- Fixed a nasty little availability bug.

v1.15; January 10, 2007

- Updated all of the new German location names.
- Thistle Tea should now only be applicable to rogues.

v1.14; December 14, 2006

- Fixed a fairly nasty startup bug.
- Updated the README.

v1.13; December 8, 2006

- Updated German capital names. (Thank you, David!)

v1.12; December 6, 2006

- Updated for LUA v5.1 (and WoW v2.0).
- Partial support for the Spanish (esES) locale.  As always,
  localization contributions are most welcome.
- "Available" recipes from the opposing faction should no longer blink
  the minimap button when entering a new region.
- Fixed the rep requirement for Enchant Weapon - Strength.

v1.11; September 18, 2006

- Fixed item linking bug (thank you, TDurden!).
- Fixed a bug that could occur for unlucky Dragonscale/Elemental LWers
  and Armorsmiths (thank you, Arilaw!).

v1.10; September 16, 2006

- I'd forgotten to enable specialization checking!  It should now work.
- Auctioneer and Recipe Book are now supported in the recipe icon
  mouseover.  Thanks very much to borohir for this functionality!

v1.9; September 5, 2006

- Recipes names are now cached when learned.  This should make new
  Blizzard patches less of a hassle.
- Availability checking now takes faction reputation and trade skill
  specialties into account.
- Adjusted some vendor information.
- Reduced memory usage significantly.

v1.8; July 11, 2006

- Several new German region names were missing in v1.7.  Hopefully they
  are all correct now.
- Adjusted some vendor information.

v1.7; June 24, 2006

- Fixed the nasty profession selection bug for non-English versions.
- Profession icons should now show up for non-English versions.
- Unlearning a profession should now be handled correctly.
- Fixed several new 1.11 German region names.
- Added the new 1.11 Cenarion Circle recipes.
- Several vendors were incorrectly marked as Alliance-only (thank you,
  Howlin!).

v1.6; June 21, 2006

- Added a right-click context menu to the scroll frame.  From this
  menu users can map vendors and expand/collapse all items.
- Fixed bug causing non-English profession stuff to suck (thank you
  ElKano and Jeremy Gurr!).
- Alts of a different team should no longer appear in the availability
  list.
- Adjusted the German translation.
- Adjusted some vendor locations.

v1.5; May 11, 2006

- All recipes that are named differently after learning (eg. transmutes)

Globals.lua

-- Globals.lua: global/saved variables and some related initialization code
-- $Id: Globals.lua 1099 2009-08-10 04:51:22Z jnmiller $
local L = LibStub("AceLocale-3.0"):GetLocale("RecipeRadar")
local BFAC = LibStub("LibBabble-Faction-3.0"):GetLookupTable()

RECIPERADAR_VERSION = "1.430"

RECIPERADAR_VENDORS_DISPLAYED = 8
RECIPERADAR_VENDOR_HEIGHT = 16
RECIPERADAR_VENDOR_OFFSET = 20
RECIPERADAR_IMAGE_ROOT = "Interface\\Addons\\RecipeRadar\\Images\\"

-- this should be <= the number of RecipeRadar*Icons
RECIPERADAR_MAPPED_VENDORS_MAX = 20

BINDING_HEADER_RECIPERADAR_HEADER = L["Recipe Radar Bindings"]
BINDING_NAME_RECIPERADAR_TOGGLE = L["Toggle Recipe Radar"]

RecipeRadar_Options = { }

RecipeRadar_Filters = {
   Teams = { },
   Professions = { },
   RealmAvails = { },
   PersonAvails = { },
}

RecipeRadar_Colors = {

   MainFont           = { r = 1.0, g = 0.75, b = 0.0 },
   Alliance           = { r = 0.5, g = 0.625, b = 1.0 },
   Horde              = { r = 1.0, g = 0.125, b = 0.25 },
   Neutral            = { r = 1.0, g = 0.82, b = 0.0 },
   LightGray          = { r = 0.9, g = 0.9, b = 0.9 },
   UncachedRecipe     = { r = 0.75, g = 0.75, b = 0.75 },
   DisabledGray       = { r = 0.5, g = 0.5, b = 0.5 },

   TooltipHeading     = { r = 0.9, g = 0.9, b = 0.9 },
   TooltipWarning     = { r = 1.0, g = 0.3, b = 0.3 },
   
   -- realm-wide availability tooltip/indicator colors
   LearnableByPlayer  = { r = 0.0, g = 1.0, b = 0.0 },
   LearnableByAlt     = { r = 0.0, g = 0.8, b = 0.0 },
   ProspectForPlayer  = { r = 1.0, g = 0.2, b = 0.2 },
   ProspectForAlt     = { r = 0.8, g = 0.2, b = 0.2 },
   KnownByPlayer      = { r = 0.7, g = 0.7, b = 0.7 },
   KnownByAlt         = { r = 0.5, g = 0.5, b = 0.5 },
   NotApplicable      = { r = 0.5, g = 0.5, b = 0.5 },
   
   -- person-wide availability tooltip/indicator colors
   Available          = GREEN_FONT_COLOR,
   Unavailable        = RED_FONT_COLOR,
   AlreadyKnown       = GRAY_FONT_COLOR,
   
   -- amount that colors will be phased to highlight the text
   Highlight          = 0.2,
   
}

RecipeRadar_Continents = {

   [0] = L["Instances"],
   [1] = L["Kalimdor"],
   [2] = L["Eastern Kingdoms"],
   [3] = L["Outland"],
   [4] = L["Northrend"],

}

RecipeRadar_Teams = {

   ["Alliance"] = {
      Name = FACTION_ALLIANCE,
      Color = RecipeRadar_Colors.Alliance,
   },
   ["Horde"] = {
      Name = FACTION_HORDE,
      Color = RecipeRadar_Colors.Horde,
   },
   ["Neutral"] = {
      Name = L["Neutral"],
      Color = RecipeRadar_Colors.Neutral,
   },

}

RecipeRadar_Professions = {

   [L["Alchemy"]] = {
      Name = L["Alchemy"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Alchemy",
   },
   [L["Blacksmithing"]] = {
      Name = L["Blacksmithing"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Blacksmithing",
   },
   [L["Cooking"]] = {
      Name = L["Cooking"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Cooking",
   },
   [L["Enchanting"]] = {
      Name = L["Enchanting"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Enchanting",
   },
   [L["Engineering"]] = {
      Name = L["Engineering"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Engineering",
   },
   [L["First Aid"]] = {
      Name = L["First Aid"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "First-Aid",
   },
   [L["Fishing"]] = {
      Name = L["Fishing"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Fishing",
   },
   [L["Leatherworking"]] = {
      Name = L["Leatherworking"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Leatherworking",
   },
   [L["Tailoring"]] = {
      Name = L["Tailoring"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Tailoring",
   },
   [L["Jewelcrafting"]] = {
      Name = L["Jewelcrafting"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Jewelcrafting",
   },
   [L["Inscription"]] = {
      Name = L["Inscription"],
      Icon = RECIPERADAR_IMAGE_ROOT .. "Icons\\" .. "Inscription",
   },

}

RecipeRadar_Specialties = {

   [L["Gnomish Engineer"]] = L["Engineering"],
   [L["Goblin Engineer"]] = L["Engineering"],
   [L["Tribal Leatherworking"]] = L["Leatherworking"],
   [L["Elemental Leatherworking"]] = L["Leatherworking"],
   [L["Dragonscale Leatherworking"]] = L["Leatherworking"],
   [L["Armorsmith"]] = L["Blacksmithing"],
   [L["Weaponsmith"]] = L["Blacksmithing"],
   [L["Master Axesmith"]] = L["Blacksmithing"],
   [L["Master Swordsmith"]] = L["Blacksmithing"],
   [L["Master Hammersmith"]] = L["Blacksmithing"],
   [L["Master Axesmith"]] = L["Blacksmithing"],
   [L["Master Swordsmith"]] = L["Blacksmithing"],
   [L["Master Hammersmith"]] = L["Blacksmithing"],
   [L["Spellfire Tailoring"]] = L["Tailoring"],
   [L["Shadoweave Tailoring"]] = L["Tailoring"],
   [L["Mooncloth Tailoring"]] = L["Tailoring"],

}

RecipeRadar_Factions = {

   BFAC["Argent Dawn"],
   BFAC["Cenarion Circle"],
   BFAC["Thorium Brotherhood"],
   BFAC["Timbermaw Hold"],
   BFAC["Zandalar Tribe"],
   BFAC["Cenarion Expedition"],
   BFAC["The Consortium"],
   BFAC["Honor Hold"],
   BFAC["Kurenai"],
   BFAC["The Mag'har"],
   BFAC["Netherwing"],
   BFAC["Sporeggar"],
   BFAC["Thrallmar"],
   BFAC["The Aldor"],
   BFAC["The Scryers"],
   BFAC["Lower City"],
   BFAC["The Sha'tar"],
   BFAC["The Violet Eye"],
   BFAC["The Scale of the Sands"],
   BFAC["Keepers of Time"],
   BFAC["Ashtongue Deathsworn"],
   BFAC["Shattered Sun Offensive"],
   BFAC["Argent Crusade"],
   BFAC["Explorers' League"],
   BFAC["Frenzyheart Tribe"],
   BFAC["Kirin Tor"],
   BFAC["Knights of the Ebon Blade"],
   BFAC["The Frostborn"],
   BFAC["The Hand of Vengeance"],
   BFAC["The Kalu'ak"],
   BFAC["The Oracles"],
   BFAC["The Silver Covenant"],
   BFAC["The Sons of Hodir"],
   BFAC["The Taunka"],
   BFAC["The Wyrmrest Accord"],
   BFAC["Valiance Expedition"],
   BFAC["Warsong Offensive"],
 --  BFAC["Guild"],

}

RecipeRadar_Availabilities = {

   ["LearnableByPlayer"] = {
      Text = L["Available Now (Player)"],
      Tooltip = { Group = 1, Heading = L["Available For:"] },
      DropDown = { POrder = 1, ROrder = 1 },
      Color = RecipeRadar_Colors.LearnableByPlayer,
      TestFunction = RecipeRadar_Availability_IsLearnableByPlayer,
   },
   ["LearnableByAlt"] = {
      Text = L["Available Now (Alts)"],
      Tooltip = { Group = 1, Heading = L["Available For:"] },
      DropDown = { POrder = nil, ROrder = 2 },
      Color = RecipeRadar_Colors.LearnableByAlt,
      TestFunction = RecipeRadar_Availability_IsLearnableByAlt,
   },
   ["ProspectForPlayer"] = {
      Text = L["Future Prospect (Player)"],
      Tooltip = { Group = 2, Heading = L["Future Prospect For:"] },
      DropDown = { POrder = 2, ROrder = 3 },
      Color = RecipeRadar_Colors.ProspectForPlayer,
      TestFunction = RecipeRadar_Availability_IsProspectForPlayer,
   },
   ["ProspectForAlt"] = {
      Text = L["Future Prospect (Alts)"],
      Tooltip = { Group = 2, Heading = L["Future Prospect For:"] },
      DropDown = { POrder = nil, ROrder = 4 },
      Color = RecipeRadar_Colors.ProspectForAlt,
      TestFunction = RecipeRadar_Availability_IsProspectForAlt,
   },
   ["KnownByPlayer"] = {
      Text = L["Already Known (Player)"],
      Tooltip = { Group = 3, Heading = L["Already Known By:"] },
      DropDown = { POrder = 3, ROrder = 5 },
      Color = RecipeRadar_Colors.KnownByPlayer,
      TestFunction = RecipeRadar_Availability_IsKnownByPlayer,
   },
   ["KnownByAlt"] = {
      Text = L["Already Known (Alts)"],
      Tooltip = { Group = 3, Heading = L["Already Known By:"] },
      DropDown = { POrder = nil, ROrder = 6 },
      Color = RecipeRadar_Colors.KnownByAlt,
      TestFunction = RecipeRadar_Availability_IsKnownByAlt,
   },
   ["NotApplicable"] = {
      Text = L["Inapplicable"],
      Tooltip = { Group = 0 },  -- do not show in tooltip
      DropDown = { POrder = 4, ROrder = 7 },
      Color = RecipeRadar_Colors.NotApplicable,
      TestFunction = nil,  -- don't need to test the default case
   },

}


-----------------------------------------------------------------------------
-- Global initialization functions
-----------------------------------------------------------------------------

-- This function "upgrades" the user's saved variables, if necessary.
function RecipeRadar_Globals_Init(force)

   RecipeRadar_Options_Init()

   if (not force and RecipeRadar_Options.Version >= RECIPERADAR_VERSION) then
      return  -- running version is current
   end

   -- v1.5 filter upgrades
   if (RecipeRadar_IsVersionLessThan(RecipeRadar_Options.Version, "1.5")) then
      if (RecipeRadar_Filters.PersonAvails[AVAILABLE]) then
         RecipeRadar_Filters.PersonAvails["LearnableByPlayer"] = true
      end
      if (RecipeRadar_Filters.PersonAvails[UNAVAILABLE]) then
         RecipeRadar_Filters.PersonAvails["ProspectForPlayer"] = true
      end
      if (RecipeRadar_Filters.PersonAvails[USED]) then
         RecipeRadar_Filters.PersonAvails["KnownByPlayer"] = true
         RecipeRadar_Filters.PersonAvails["NotApplicable"] = true
      end
      RecipeRadar_Filters.PersonAvails[AVAILABLE] = nil
      RecipeRadar_Filters.PersonAvails[UNAVAILABLE] = nil
      RecipeRadar_Filters.PersonAvails[USED] = nil
   end
   
   -- v1.9 skill DB format change
   if (RecipeRadar_IsVersionLessThan(RecipeRadar_Options.Version, "1.9") and
         RecipeRadar_SkillDB) then
      for realm, realm_info in pairs(RecipeRadar_SkillDB) do
         if (type(realm_info) == "string") then
            realm_info = nil
         else
            realm_info.Team = nil
            for player, player_info in pairs(realm_info) do
               local temp = { }
               for prof, prof_info in pairs(player_info) do
                  if (RecipeRadar_IsProfession(prof)) then
                     temp[prof] = prof_info
                     player_info[prof] = nil
                  end
               end
               if (not player_info.Professions) then
                  player_info.Professions = { }
               end
               for k, v in pairs(temp) do
                  player_info.Professions[k] = v
               end
            end
         end
      end
   end

   -- v1.12 skill DB version removal (sigh)
   if (RecipeRadar_SkillDB and RecipeRadar_SkillDB.Version) then
      RecipeRadar_SkillDB.Version = nil
   end

   -- v1.15 (now v1.17 thanks to my dumbness) skill DB cleanup
   if (RecipeRadar_IsVersionLessThan(RecipeRadar_Options.Version, "1.17") and
         RecipeRadar_SkillDB) then
      for realm, realm_info in pairs(RecipeRadar_SkillDB) do
         for player, player_info in pairs(realm_info) do
            if (player_info.Professions) then
               for prof, _ in pairs(player_info.Professions) do
                  if (not RecipeRadar_IsProfession(prof)) then
                     player_info.Professions[prof] = nil
                  end
               end
            end
         end
      end
   end

   -- upgrade is complete
   RecipeRadar.db.profile.Version = RECIPERADAR_VERSION

end

-- Initializes the Options table with some defaults.
function RecipeRadar_Options_Init()

   RecipeRadar_Options.Version = RECIPERADAR_VERSION

   if (RecipeRadar.db.profile.Version == nil) then
      -- setting this to the lowest version forces an
      -- upgrade for all characters on this account
      RecipeRadar.db.profile.Version = "1.0"
   end

   if (RecipeRadar.db.profile.MinimapButtonPosition == nil) then
      RecipeRadar.db.profile.MinimapButtonPosition = 256
   end

   if (RecipeRadar.db.profile.ShowMinimapButton == nil) then
      RecipeRadar.db.profile.ShowMinimapButton = true
      RecipeRadarMinimapButtonFrame:Show()
   end

   if (not RecipeRadar_IsProfession(
         RecipeRadar.db.profile.CurrentProfession)) then
      RecipeRadar.db.profile.CurrentProfession = L["Cooking"]
   end

   if (RecipeRadar.db.profile.RealmAvailability == nil) then
      RecipeRadar.db.profile.RealmAvailability = true
      RecipeRadar_RealmAvailDropDown:Show()
   end

   if (RecipeRadar.db.profile.Locked == nil) then
      RecipeRadar.db.profile.Locked = false
   end

   if (RecipeRadar.db.profile.AutoSelectRegion == nil) then
      RecipeRadar.db.profile.AutoSelectRegion = true
   end

   if (RecipeRadar.db.profile.AutoMapVendors == nil) then
      RecipeRadar.db.profile.AutoMapVendors = false
   end

end
